### PR TITLE
SUS-1782: Cleanup category table

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-1782

### CODE REVIEW

File: includes/Category.php [hunk 1]
```diff
@@ -72,7 +80,17 @@ protected function initialize() {
 		# and should not be kept, and 2) we *probably* don't have to scan many
 		# rows to obtain the correct figure, so let's risk a one-time recount.
 		if ( $this->mPages < 0 || $this->mSubcats < 0 || $this->mFiles < 0 ) {
-			$this->refreshCounts();
```
Comment (macbre) [inline includes/Category.php:75]: :+1: for moving it to an offline task

File: includes/wikia/tasks/Tasks/RefreshCategoryCountsTask.php [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	private $db;
+
+	public function __construct() {
+		$this->db = wfGetDB( DB_MASTER );
+	}
+
+	/**
+	 * @param string[]|string $data title of category or array of category titles
+	 */
+	public function refreshCounts( $data ) {
```
Comment (macbre) [inline includes/wikia/tasks/Tasks/RefreshCategoryCountsTask.php]: `s/$data/$categoryNames/g` ?

File: maintenance/cleanupEmptyCategories.php [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+script is perfectly safe to run on large, live wikis, and running it multiple
+times is harmless. You may want to use the throttling options if it's causing
+too much load; they will not affect correctness.
+
+If the script is stopped and later resumed, you can use the --mode and --begin
+options with the last printed progress indicator to pick up where you left off.
+
+When the script has finished, it will make a note of this in the database, and
+will not run again without the --force option.
+TEXT
```
Comment (macbre) [inline maintenance/cleanupEmptyCategories.php:50]: :heart:

**Review by macbre — APPROVED** [review]

**Comment by mszabo-wikia** [comment]
I will prepare a new PR for this ticket.

File: maintenance/cleanupEmptyCategories.php [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+			true
+		);
+		$this->addOption( 'dry-run', 'Only list affected categories, don\'t make changes', false, false );
+	}
+
+	protected function getUpdateKey() {
+		return 'cleanup empty categories';
+	}
+
+	protected function doDBUpdates() {
```
Comment (WikiaTech) [inline maintenance/cleanupEmptyCategories.php]: MAJOR The Cyclomatic Complexity of this function "doDBUpdates" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
MAJOR This function "doDBUpdates" has 153 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: maintenance/cleanupEmptyCategories.php (hunk 4 above)
Comment (WikiaTech) [inline maintenance/cleanupEmptyCategories.php:79]: MAJOR The Cyclomatic Complexity of this function "doDBUpdates" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
MAJOR This function "doDBUpdates" has 153 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: includes/CategoryViewer.php [hunk 5]
```diff
@@ -700,15 +700,10 @@ private function getCountMessage( $rescnt, $dbcnt, $type ) {
 			# quick due to the small number of entries.
 			$totalcnt = $rescnt;
 
-			// Wikia change - begin
-			// @see SUS-2050
-			global $wgCityId;
-
-			$task = new \Wikia\Tasks\Tasks\CategoryRefreshCountsTask();
```
Comment (macbre) [inline includes/CategoryViewer.php:707]: The old `CategoryRefreshCountsTask` task has been deleted as well

**Pull request merged**